testcases/kernel/syscalls/write/write05.c

```diff
@@ -57,160 +57,127 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <sys/wait.h>
 #include <sys/mman.h>
-#include "test.h"
+#include "tst_test.h"
 
-void setup(void);
-void cleanup(void);
-
-char *TCID = "write05";
-int TST_TOTAL = 1;
 char filename[100];
 int fd;
 
 char *bad_addr = 0;
 
-int main(int argc, char **argv)
+static void verify_write(void)
 {
-	int lc;
-
 	char pbuf[BUFSIZ];
 	int pipefildes[2];
 	int status, pid;
 
-	tst_parse_opts(argc, argv, NULL, NULL);
-
-	/* global setup */
-	setup();
-
-	/* The following loop checks looping state if -i option given */
-	for (lc = 0; TEST_LOOPING(lc); lc++) {
-
-		/* reset tst_count in case we are looping */
-		tst_count = 0;
-
 //block1:
-		tst_resm(TINFO, "Enter Block 1: test with bad fd");
-		if (write(-1, pbuf, 1) != -1) {
-			tst_resm(TFAIL, "write of invalid fd passed");
-		} else {
-			if (errno != EBADF) {
-				tst_resm(TFAIL, "expected EBADF got %d", errno);
-			}
-			tst_resm(TPASS, "received EBADF as expected.");
+	tst_res(TINFO, "Enter Block 1: test with bad fd");
+	if (write(-1, pbuf, 1) != -1) {
+		tst_res(TFAIL, "write of invalid fd passed");
+	} else {
+		if (errno != EBADF) {
+			tst_res(TFAIL, "expected EBADF got %d", errno);
 		}
-		tst_resm(TINFO, "Exit Block 1");
+		tst_res(TPASS, "received EBADF as expected.");
+	}
+	tst_res(TINFO, "Exit Block 1");
 
 //block2:
-		tst_resm(TINFO, "Enter Block 2: test with a bad address");
-		fd = creat(filename, 0644);
-		if (fd < 0) {
-			tst_resm(TFAIL, "creating a new file failed");
-			cleanup();
-		}
-		if (write(fd, bad_addr, 10) != -1) {
-			tst_resm(TFAIL, "write() on an invalid buffer "
-				 "succeeded, but should have failed");
-			cleanup();
-		} else {
-			if (errno != EFAULT) {
-				tst_resm(TFAIL, "write() returned illegal "
-					 "errno: expected EFAULT, got %d",
-					 errno);
-				cleanup();
+	tst_res(TINFO, "Enter Block 2: test with a bad address");
+	fd = creat(filename, 0644);
+	if (fd < 0)
+		tst_res(TFAIL, "creating a new file failed");
+
```

Comment:
Here as well, we should make use of the SAFE_CREAT(), also notice that tst_res() is wrong here, since we want to exit the test with tst_brk(). Moreover this kind of initialization should be done once in the test setup anyway.